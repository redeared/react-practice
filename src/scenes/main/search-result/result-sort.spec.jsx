import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import { ResultSort } from './result-sort';

describe('ResultSort', () => {
    it('should rendered without chash', () => {
        const comp = create(<ResultSort
            sortOrder={{
                sortByRating: false,
                sortByReleaseDate: false,
            }}
            changeSortOrder={() => 'empty'}
        />).toJSON();
        expect(comp).toMatchSnapshot();
    });
    it('should call changeSortOrder with invert sortByRating option when click on "rating"', () => {
        let sortByRating = false;

        const comp = shallow(<ResultSort
            sortOrder={{
                sortByRating,
                sortByReleaseDate: false,
            }}
            changeSortOrder={(value) => { ({ sortByRating } = value); }}
        />);

        const ratingBlock = comp.findWhere(n => n.props().children === 'rating');
        ratingBlock.simulate('click');
        expect(sortByRating).toBe(true);
    });
});
