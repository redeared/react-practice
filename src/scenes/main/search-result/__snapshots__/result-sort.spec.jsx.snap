// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultSort should rendered without chash 1`] = `
Array [
  "Sort by:",
  " ",
  <span
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={null}
    tabIndex="0"
  >
    release date
  </span>,
  " ",
  <span
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={null}
    tabIndex="0"
  >
    rating
  </span>,
]
`;
